modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala

```diff
@@ -156,10 +156,16 @@ object SharedOptionsUtil extends CommandHelpers {
       jmhVersion: Option[String] = None,
       ignoreErrors: Boolean = false
     ): bo.BuildOptions = {
-      val platformOpt =
-        if (js.js) Some(Platform.JS)
-        else if (native.native) Some(Platform.Native)
-        else None
+      val platformStr = Platform.normalize(platform.getOrElse("None"))
+      val platformOpt = Platform.parse(platformStr) match {
+        case Some(p) => Some(p)
+        case None =>
+          if (native.native && js.js) Some(Platform.Native)
+          else if (js.js) Some(Platform.JS)
+          else if (native.native) Some(Platform.Native)
+          else None
+      }
```

Comment:
To avoid executing the following expression `platform.getOrElse("None")`, you can do something like this:

```suggestion
      val platformStr = platform.map(Platform.normalize)
      val platformOpt = platformStr.map(Platform.parse) match {
        case Some(p) => Some(p)
        case None =>
          if (native.native && js.js) Some(Platform.Native)
          else if (js.js) Some(Platform.JS)
          else if (native.native) Some(Platform.Native)
          else None
      }
```